File: frontend/lib/auth/models/register_request.dart
class RegisterRequest {
  final String email;
  final String password;
  final String? inviteCode;
  final String verificationCode;

  RegisterRequest({
    required this.email,
    required this.password,
    required this.verificationCode,
    this.inviteCode,
  });

  Map<String, dynamic> toJson() {
    final code = inviteCode;
    return {
      'email': email,
      'password': password,
      'verification_code': verificationCode,
      if (code != null && code.isNotEmpty) 'invite_code': code,
    };
  }
}

File: backend/src/main/java/top/kariscode/karisreview/auth/dto/RegisterRequest.java
package top.kariscode.karisreview.auth.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public class RegisterRequest {

    @NotBlank(message = "{validation.email.notblank}")
    @Email(message = "{validation.email.invalid}")
    private String email;

    @NotBlank(message = "{validation.password.notblank}")
    @Size(min = 6, max = 128, message = "{validation.password.length}")
    private String password;

    private String inviteCode;

    /** 注册邮箱验证码（注册必须先通过 /auth/register-code 获取；缺失由 service 层检查，保证邀请码校验优先）。 */
    private String verificationCode;

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    public String getPassword() { return password; }
    public void setPassword(String password) { this.password = password; }
    public String getInviteCode() { return inviteCode; }
    public void setInviteCode(String inviteCode) { this.inviteCode = inviteCode; }
    public String getVerificationCode() { return verificationCode; }
    public void setVerificationCode(String verificationCode) { this.verificationCode = verificationCode; }
}